### PR TITLE
W-12050227: Adding cluster components extension model

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

File: runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java [hunk 1]
```diff
@@ -80,13 +80,13 @@
 import static org.mule.runtime.internal.dsl.DslConstants.CORE_PREFIX;
 import static org.mule.runtime.internal.dsl.DslConstants.CORE_SCHEMA_LOCATION;
 import static org.mule.runtime.internal.dsl.DslConstants.FLOW_ELEMENT_IDENTIFIER;
+import static org.mule.sdk.api.stereotype.MuleStereotypes.CONFIGURATION_ELEMENT;
 
+import static com.google.common.collect.ImmutableSet.of;
```
Comment (elrodro83) [inline runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java:85]: java imports go first

File: runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java [hunk 2]
```diff
@@ -1122,6 +1123,12 @@ private void declareConfiguration(ExtensionDeclarer extensionDeclarer) {
         .describedAs("The default correlation id generation expression for every source. This must be DataWeave expression.");
   }
 
+  private void declareExtraAppConfigs(ConstructDeclarer configuration) {
```
Comment (elrodro83) [inline runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java:1126]: rename

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java (hunk 1 above)
Comment (eze210) [reply runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java]: done

File: runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java (hunk 2 above)
Comment (eze210) [reply runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java]: done, also added a comment to be clearer on what does it represent

File: runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java [hunk 3]
```diff
@@ -1122,6 +1124,13 @@ private void declareConfiguration(ExtensionDeclarer extensionDeclarer) {
         .describedAs("The default correlation id generation expression for every source. This must be DataWeave expression.");
   }
 
+  // See the "abstract-configuration-element" in the mule-core-common.xsd file. It represents the same placeholder.
+  private void declareConfigurationElementsPlaceholder(ConstructDeclarer configuration) {
+    configuration.withOptionalComponent("abstractConfigurationElement")
```
Comment (elrodro83) [inline runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java:1129]: add the value of the xsd (https://github.com/mulesoft/mule/blob/master/runtime-extension-model/src/main/resources/META-INF/mule-core-common.xsd#L650) as the description, so the source of truth is the extModel

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**